thewalrus/decompositions.py

```diff
@@ -218,7 +218,7 @@ def takagi(A, svd_order=True, rtol=1e-16):
         return l, U
 
     u, d, v = np.linalg.svd(A)
```

Comment:
I think it's important to remember that `numpy.linalg.svd` returns the hermitian conjugate of V (as in Eq. (34) of the paper).
```suggestion
    u, d, vh = np.linalg.svd(A)
```
Thus, I'm not convinced that either of these expressions matches the expression in Box 1. There the desired expression is (U^T V)* = (V^T U)* -> `vh @ u.conj()`. What was previously written on line 221 has an extra transpose (`.T`), and the new expression on line 221 has not taken into account the relationship between V and vh.

Reply:
> I think it's important to remember that `numpy.linalg.svd` returns the hermitian conjugate of V (as in Eq. (34) of the paper).
> 
> Thus, I'm not convinced that either of these expressions matches the expression in Box 1. There the desired expression is (U^T V)* = (V^T U)* -> `vh @ u.conj()`. What was previously written on line 221 has an extra transpose (`.T`), and the new expression on line 221 has not taken into account the relationship between V and vh.

what you call the last matrix in the SVD is immaterial, it is just that, the last unitary:

<img width="453" height="315" alt="image" src="https://github.com/user-attachments/assets/04b2f784-b31f-4484-b797-6e444f8efa10" />

Reply:
Hmm... if all of these products are equivalent, but `sqrtm` doesn't seem to treat them as such, then indeed something funny seems to be going on.